src/Mutator/NodeMutationGenerator.php

```diff
@@ -130,11 +129,7 @@ private function generateForMutator(Node $node, IgnoreMutator $mutator, array $m
             return $mutations;
         }
 
-        $mutatedResult = $mutator->mutate($node);
-
-        $mutatedNodes = $mutatedResult instanceof Generator ? $mutatedResult : [$mutatedResult];
-
-        foreach ($mutatedNodes as $mutationByMutatorIndex => $mutatedNode) {
+        foreach ($mutator->mutate($node) as $mutationByMutatorIndex => $mutatedNode) {
```

Comment:
This is certainly simpler.

Here is another small catch, which is absent with arrays: generators can have multiple values for the same key.

Reply:
Caught that after reviewing a bit more in details your comment. I added an `iterator_to_array($x, false)` with a comment about it. I would like to cover it in a test but unfortunately `NodeMutationGenerator` is completely untested at the moment